lib/util.js

```diff
@@ -152,6 +152,51 @@ async function streamToBuffer(method, url, status, stream, totalSize) {
     });
 }
 
+/**
+ * Converts the content of a readable Stream into a Buffer
+ * instance. WARNING: this method will read the entirety of
+ * the stream into memory; use with extreme caution.
+ *
+ * @param {String} method HTTP method
+ * @param {String} url URL requested
+ * @param {Number} status HTTP status code of response
+ * @param {ReadableStream} stream Stream to convert.
+ * @param {number} totalSize Total size of the stream.
+ * @returns {Promise<TransferMemoryBlock>} Contents of the stream in a
+ *  pooled buffer memory block.
+ */
+async function streamToPooledBuffer(method, url, status, stream, totalSize, bufferBlockMemoryAllocator) {
+    if (!bufferBlockMemoryAllocator) {
+        return streamToBuffer(method, url, status, stream, totalSize);
```

Comment:
nth: add log to let user know it is using the older method without the mem allocator

Reply:
Actually the opposite I think - let the user (developer) know they're using a `streamToBuffer` backed by a pooled buffer (and the memory allocator).